### PR TITLE
[ranges] Shorten and adjust stable labels

### LINKED ISSUE #2513: revise section labels introduced by ranges
In particular, [range] should be [ranges] for consistency with the rest of the library. There are probably other opportunities for cleanups here.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2513.

### CODE REVIEW

**Comment by jensmaurer** [comment]
@CaseyCarter, I'll try [common.alg.req.ind.cmp] etc.

**Comment by jensmaurer** [comment]
Regarding [iterator.concept.weaklyincrementable]: Regrettably, none of the other stable labels in the vicinity abbreviate "iterator" as "iter". I've abbreviate the stuff at the end a bit.

File: source/iterators.tex [hunk 1]
```diff
@@ -93,19 +93,19 @@
...
+  // \ref{iterator.concept.inc}, concept \tcode{Incrementable}
   template<class I>
     concept Incrementable = @\seebelow@;
 
   // \ref{iterator.concept.iterator}, concept \tcode{Iterator}
   template<class I>
     concept Iterator = @\seebelow@;
 
-  // \ref{iterator.concept.incrementable}, concept \tcode{Sentinel}
+  // \ref{iterator.concept.sentinel}, concept \tcode{Sentinel}
```
Comment (CaseyCarter) [inline source/iterators.tex]: Good catch.

**Comment by CaseyCarter** [comment]
> Do we even need "common" in front? [alg.req...] seems good enough.

[alg.req] could be easily confused with [algorithms.requirements], but I'm not concerned that this will be a problem in practice since [alg.req] contains only subclauses and [algorithms.requirements] contains only paragraphs.

Thanks for the cleanup.

**Pull request merged**